include/exadg/operators/inverse_mass_operator.h

```diff
@@ -175,15 +237,39 @@ class InverseMassOperator
   void
   apply(VectorType & dst, VectorType const & src) const
   {
-    dst.zero_out_ghost_values();
-
-    if(data.implementation_type == InverseMassType::MatrixfreeOperator)
+    if(data.implementation_type == InverseMassType::GlobalKrylovSolver)
     {
-      matrix_free->cell_loop(&This::cell_loop_matrix_free_operator, this, dst, src);
```

Comment:
note how here we did this also for `&src == &dst` which was not save for the `ElementwiseKrylov`